crates/nu-command/src/conversions/into/binary.rs

```diff
@@ -209,6 +206,32 @@ pub fn action(input: &Value, _args: &Arguments, span: Span) -> Value {
                 src_span: other.expect_span(),
             }),
         },
+    };
+
+    if _args.compact {
+        if let Value::Binary { val, span } = value {
+            let val = if cfg!(target_endian = "little") {
+                let mut idx = val.len() - 1;
+                while val[idx] == 0 {
+                    idx -= 1;
+                }
+                let (val, _) = val.split_at(idx + 1);
```

Comment:
This `val.split_at` surprised me. It looks like we're just taking the last 0 off until there are no more zeros. This makes my "spidey sense" go off as not being the right way to handle this, but I could be wrong.